### PR TITLE
Preserve complete standard UM records

### PR DESCRIPTION
**Pull request by miyamamoto**
## Summary

- add the official 10-digit `KettoNum` identity and primary key to standard `UMA`
- preserve `DelDate=00000000`, expand all 27 x 6 placing counts plus four running-style counts, and store `TorokuRaceSu`
- reject malformed UM identities and unsafe legacy constraints before schema or row mutation
- document the required backup/rebuild/reimport boundary for old standard `UMA` tables

## Root cause and impact

The standard table omitted the official registration key, so valid UM records could not be stored safely. The first key repair also exposed that the parser's compact body was not translated to the expanded standard columns: imports reported success while 167 official leaves were dropped. Existing additional `UNIQUE` constraints could also change replacement semantics and silently remove a different horse.

The repair validates `KettoNum` as exactly ten ASCII digits, expands every compact field with exact cardinality, and preflights additional UNIQUE/exclusion constraints and PostgreSQL primary-key usability before additive migration. Existing keyless/wrong-key tables are never inferred or mutated; operators must back up, recreate, and reimport.

## Red-first evidence

Against the pre-repair candidate, the combined focused selection produced 22 failures; a malformed compact-body record was also incorrectly imported. Independent review then mutated the expander to copy first-place into places 2-6: the old repeated sentinel fixture stayed green. The final fixture uses distinct ordered values `001` through `162`, and the same mutation now fails all complete-translation, normal-batch, optimized-batch, and single-record contracts.

## Validation

Candidate full SHA: `6cd699aa3bc9b82b902e06be2625243a02ad1e33`

- full local suite: `2823 passed, 136 skipped, 22 subtests passed`
- final UM focused: `127 passed, 5 skipped`
- fresh PostgreSQL 16: all 5 UM integration contracts passed after the final sentinel update
- affected UM/migration/WF constraint selection: `164 passed, 6 skipped, 103 deselected`
- `uv lock --check`
- fail-closed test gate, compileall, fatal flake8, strict MkDocs, `git diff --check`
- exact `git archive` wheel/sdist: distribution-content gate and extracted-wheel init/SQLite smoke passed; `specs/` excluded
- two independent Codex reviews: final `P0/P1/P2 = 0/0/0`, exact SHA and clean worktree

This is one release-readiness iteration only. It does not publish a release; the remaining key/erase-contract iterations continue separately.

## Summary by CodeRabbit

- **New Features**
  - Improved standard UM/UMA imports with expanded horse-placement and training-count fields.
  - Added support for 10-digit registration numbers and lossless preservation of deletion dates, including `00000000`.
  - Added stricter validation for registration keys and incompatible database schemas.

- **Bug Fixes**
  - Corrected UMA field mapping and schema definitions.
  - Improved update and migration safety to prevent unintended data changes.

- **Documentation**
  - Documented supported formats, schema requirements, validation rules, and recovery procedures.

### CODE REVIEW

File: tests/test_um_parser_layout.py [hunk 1]
```diff
@@ -161,3 +204,478 @@ def test_exact_1609_byte_crlf_record_is_accepted(self):
...
+        database.execute(f"CREATE SCHEMA {schema_name}")
+        database.execute(f"SET search_path TO {schema_name}")
+        database.commit()
+        yield database
+    finally:
+        try:
+            database.execute(f"DROP SCHEMA IF EXISTS {schema_name} CASCADE")
+            database.commit()
+        finally:
+            database.disconnect()
```
[the remark this answers is not in the document]
Comment (miyamamoto) [reply tests/test_um_parser_layout.py:501]: Thank you for catching this teardown edge case. Commit 438c4bb0661286c3dda66ceb17ca59b930dd6f07 now rolls the PostgreSQL fixture back before dropping its temporary schema, so an aborted test transaction cannot block cleanup. The same review-response batch also parameterizes the DualDatabase regression across both primary and secondary unsafe targets. The complete UM test file passed 133 tests against SQLite and a fresh PostgreSQL 16 instance; production source is unchanged.

**Comment by miyamamoto** [comment]
Thank you for the review. I addressed both test-only suggestions together in full SHA `438c4bb0661286c3dda66ceb17ca59b930dd6f07`: the DualDatabase unsafe-UNIQUE contract now covers both primary and secondary targets, and the PostgreSQL fixture rolls back before schema cleanup. Verification: the complete UM contract passed **133 tests** against SQLite and a fresh PostgreSQL 16 instance; `uv lock --check`, the test gate, compileall, fatal flake8, strict MkDocs, and `git diff --check` also passed. No production source changed in this response commit.
